samples/peripheral/802154_phy_test/src/comm_proc.c

```diff
@@ -6,32 +6,39 @@
 /* Purpose: Communication interface for library */
 
 #include <stdint.h>
-
 #include <zephyr/shell/shell.h>
-
 #include <zephyr/logging/log.h>
-LOG_MODULE_REGISTER(comm);
-
 #include "comm_proc.h"
 #include "ptt_uart_api.h"
 
+LOG_MODULE_REGISTER(comm);
+
 static char cmd_buf[COMM_MAX_TEXT_DATA_SIZE];
 static const struct shell *shell;
 
-static int cmd_custom(const struct shell *shell, size_t argc, char **argv)
+static int cmd_custom(const struct shell *shell_inst, size_t argc, char **argv)
 {
-	size_t len = 0;
+	size_t max_append_len = 0;
 
 	memset(cmd_buf, 0, sizeof(cmd_buf));
 
 	for (int i = 0; i < argc; i++) {
-		len = COMM_MAX_TEXT_DATA_SIZE - strlen(cmd_buf);
-		if (len < strlen(argv[i])) {
-			shell_error(shell, "the command is too long");
+		/* cmd_buf should contain space for the command and a NULL character.
+		 * Hence we reduce 1 here to compute the max_append_len of the maximum allowed
+		 * length of string that can be appended.
+		 */
+		max_append_len = COMM_MAX_TEXT_DATA_SIZE - strlen(cmd_buf) - 1;
+		if (i < (argc - 1)) {
+			/* When appending the last command, we need to ensure that we have
+			 * space for an additional space character.
```

Comment:
nit: The comment mentions appending the last command, but the `if` statement is executed on anything but the last `argv`. I think this should be rephrased.